templates/contribute/naming.md

```diff
@@ -15,7 +15,7 @@ in mathlib under Lean 4 we use a combination of `snake_case`, `lowerCamelCase` a
 3. Functions are named the same way as their return values (e.g. a function of type `A → B → C` is named as though it is a term of type `C`).
 4. All other terms of `Type`s (basically anything else) are in `lowerCamelCase`.
 5. When something named with `UpperCamelCase` is part of something named with `snake_case`, it is referenced in `lowerCamelCase`.
-6. Acronyms like `LE` are written upper-/lowercase as a group, depending on what the first character would be.
+6. Acronyms like `LE` are written upper-/lowercase as a group, depending on what the first character would be. When they occur as fields of some structure they are normally lowercase.
```

Comment:
What I mean here is that the first letter of the unique field of `LT` should be capitalized and hence this rule says it should be `LT.LT` but it is `LT.lt`.

Reply:
[Link](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/naming.20conventions.3A.20LT.2Elt/near/434941863) to the discussion on Zulip and [another](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Naming.20convention/near/395698303) to the one before that.

@faenuccio  I am not sure I understand your comment immediately above. Would I be correct in thinking that you find `LT.lt` preferable for aesthetic reasons (as presumably did Leo when he wrote it three years ago) and are willing to make the rule set more complex to accommodate it?

Reply:
I am not find this preferable, but I am describing the current situation in Mathlib, see for instance [Preorder](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Init/Order/Defs.html#Preorder) or [Btw](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Circular.html#Btw). I think that the guideline is inaccurate and I find it important to make it adhere to what happens in mathlib so that new contributors can find their way, I do not go as far as criticising Leo's choices.

Reply:
Well then I guess the question is whether the naming conventions should be regarded as prescription or description.

Reply:
Well, I believe that the guide style should certainly be aligned with the current situation. So I think that it is correct that for the time being it describes how to read the library and how to align new definitions to it. If we decide to change the naming convention, this would entail a change of the guide style but having a guide style that contradicts the library does not seem optimal.

Reply:
The current situation is that Prop-valued fields are capitalised, except for a few ones in Core that we can't change. This does *not* match what you are suggesting to change the naming convention to.

Reply:
For some reason I have the feeling that either I do not understand or I cannot make my point. I am not suggesting any change to the convention whatsoever. I am suggesting that we change the doc because it is not aligned with the convention.

Take [Btw](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Circular.html#Btw), which is not in Core. Its unique field is
```lean
btw : α → α → α → [Prop](https://leanprover-community.github.io/mathlib4_docs/foundational_types.html)
```
which is not capitalized but that it should be precisely because of what you are saying. This PR aims at adding these kind of things in the list of exceptions mentioned in 6.

Reply:
"When they occur as fields of some structure they are normally lowercase." is still a prescriptive sentence that is wrong.

`Btw.btw` is indeed another exception. The reason is that the current naming conventions suggest the name `Btw.Btw`. But then the class and the field have the same name, which is problematic. So it should probably be `BTW.Btw` or something.

Reply:
If you want to add these exceptions, make it something like "Rule 2 has some exceptions: some fields of structures are currently wrongly lower-cased [insert examples]."

Note that this is *not* an exception to rule 6, but to rule 2.

Reply:
> But then the class and the field have the same name, which is problematic

Is it?

Reply:
If the next line is `[export Btw (Btw)](https://github.com/leanprover-community/mathlib4/blob/b6cc5394d8cb9203e29724f04d35fbcdf24b785a/Mathlib/Order/Circular.lean#L97)`, then I'd say yes.

Reply:
Hmm maybe that means we need to use notation for it

Reply:
OK, I have added more remarks to rule 2 and said that `LT.lt` is an exception.

Reply:
```suggestion
6. Acronyms like `LE` are written upper-/lowercase as a group, depending on what the first character would be. 
```